plugins/backstage-plugin-coder/src/components/CoderProvider/CoderAuthProvider.tsx

```diff
@@ -331,6 +409,218 @@ function generateAuthState({
   };
 }
 
-function readAuthToken(): string {
-  return window.localStorage.getItem(TOKEN_STORAGE_KEY) ?? '';
+// Have to get the root of the React application to adjust its dimensions when
+// we display the fallback UI. Sadly, we can't assert that the root is always
+// defined from outside a UI component, because throwing any errors here would
+// blow up the entire Backstage application, and wreck all the other plugins
+const mainAppRoot = document.querySelector<HTMLElement>(BACKSTAGE_APP_ROOT_ID);
+
+type StyleKey = 'landmarkWrapper' | 'dialogButton' | 'logo';
+type StyleProps = Readonly<{ isDialogOpen: boolean }>;
+
+const useFallbackStyles = makeStyles<Theme, StyleProps, StyleKey>(theme => ({
+  landmarkWrapper: ({ isDialogOpen }) => ({
+    zIndex: isDialogOpen ? 0 : 9999,
+    position: 'fixed',
+    bottom: theme.spacing(2),
+    width: '100%',
+    maxWidth: 'fit-content',
+    left: '50%',
+    transform: 'translateX(-50%)',
+  }),
+
+  dialogButton: {
+    display: 'flex',
+    flexFlow: 'row nowrap',
+    columnGap: theme.spacing(1),
+    alignItems: 'center',
+  },
+
+  logo: {
+    fill: theme.palette.primary.contrastText,
+    width: theme.spacing(3),
+  },
+}));
+
+function FallbackAuthUi() {
+  /**
+   * Add additional padding to the bottom of the main app to make sure that even
+   * with the fallback UI in place, it won't permanently cover up any of the
+   * other content as long as the user scrolls down far enough.
+   *
+   * Involves jumping through a bunch of hoops since we don't have 100% control
+   * over the Backstage application. Need to minimize risks of breaking existing
+   * Backstage styling or other plugins
+   */
+  const fallbackRef = useRef<HTMLElement>(null);
+  useLayoutEffect(() => {
```

Comment:
The things you do to avoid breaking other people's code

Reply:
lol